serverless-scheduler-proxy/main.go

```diff
@@ -88,45 +89,70 @@ func main() {
 
 func rewriteBotCronURL(c botConfig) func(*http.Request) {
 	return func(req *http.Request) {
-
-		var bodyBytes []byte
-		if req.Body != nil {
-			bodyBytes, _ = ioutil.ReadAll(req.Body)
-			req.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
-		} else {
-			log.Println("request had no body")
+		req.Header.Add("x-github-event", "schedule.repository")
+		parser := func(bodyBytes []byte) (string, string) {
+			var pay reqPayload
+			json.Unmarshal(bodyBytes, &pay)
+			return pay.Name, pay.Location
 		}
+		rewriteBotURL(c, parser, req)
+	}
+}
 
-		var pay reqPayload
-		json.Unmarshal(bodyBytes, &pay)
+func rewriteBotPubSubURL(c botConfig) func(*http.Request) {
+	return func(req *http.Request) {
+		req.Header.Add("x-github-event", "pubsub.message")
+		parser := func(bodyBytes []byte) (string, string) {
+			var pay PubSubMessage
+			json.Unmarshal(bodyBytes, &pay)
+			log.Printf("handling pubsub message for subscription: %v\n", pay.Subscription)
+
```

Comment:
I"m pretty sure we need to find a way to insert an `installation` identifier, perhaps this can be set as the job is dispatched? what are you thinking.